privacyIDEAADFSProvider/PrivacyIDEA.cs

```diff
@@ -87,54 +92,103 @@ public PIResponse ValidateCheckWebAuthn(string user, string transactionid, strin
             {
                 Log("ValidateCheckWebAuthn called with missing parameter: user=" + user + ", transactionid=" + transactionid
                     + ", WebAuthnSignResponse=" + webAuthnSignResponse + ", origin=" + origin);
+                return null;
             }
 
-            var parameters = new List<KeyValuePair<string, string>>();
-            parameters.Add(new KeyValuePair<string, string>("user", user));
-            parameters.Add(new KeyValuePair<string, string>("pass", ""));
-            parameters.Add(new KeyValuePair<string, string>("transaction_id", transactionid));
+            // Parse the WebAuthnSignResponse and add mandatory parameters
+            JToken root;
+            try
+            {
+                root = JToken.Parse(webAuthnSignResponse);
+            }
+            catch (JsonReaderException jex)
+            {
+                Error("WebAuthnSignRequest does not have the required format! " + jex.Message);
+                return null;
+            }
 
-            var headers = new List<KeyValuePair<string, string>>();
-            headers.Add(new KeyValuePair<string, string>("Origin", origin));
+            string credentialid = (string)root["credentialid"];
+            string clientdata = (string)root["clientdata"];
+            string signaturedata = (string)root["signaturedata"];
+            string authenticatordata = (string)root["authenticatordata"];
 
+            var paramDict = new Dictionary<string, string>
+            {
+                { "user", user },
+                { "pass", "" },
```

Comment:
can we have a "realm" here as parameter?

Reply:
i have omitted the realm since the next feature i am going to implement will be the realm mapping which requires some more logic than to just add the realm.

Reply:
OK, this is the webauthn PR. So realms will be handled later. THanks.